### ISSUE TITLE
Update ImageView falling infinite loop on using sd_setImage on layoutSubviews

### ISSUE THREAD

**Issue by mqzkim** [rs/SDWebImage#2097]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 11.2
 SDWebImage Version      | 4.2.1
 Integration Method      | cocoapods
 Xcode Version           | Xcode 9.2 beta
 Repro rate              | all the time (100%)
 Repro with our demo prj | Yes
 Demo project link       | https://github.com/mqzkim/DemoProjectForSDWebImageTest

### Issue Description and Steps

I made the app update image view( like icon ) using `[sd_setImageWithURL:]` on `layoutSubviews()`
from `UICollectionViewCell`, `UITableViewCell`, `UIViewCell`.
On SDWebImage v4.1.2, it updates image views easily.

    override func layoutSubviews() {
        super.layoutSubviews()
        
        iconImageView.sd_setImage(with: URL(string: IMAGE_URL))
    }
After I updated SDWebImage to v4.2.1, the `UICollectionViewCell` doesn't update image.
Carefully, `layoutSubviews` is running infinitely..

So I made Demo project with `UICollectionViewCell`, `UITableViewCell`, and `UIViewCell`. You can see that only `UICollectionViewCell` doesn't update the icon view.

**Comment by dreampiggy** [comment]
@mqzkim This is because of set-image block will schedule on next runloop, and for this demo project, you use autolayout and add a constrains to cell, this will trigger a loop call.

This bug has been fixed on the master branch but not currently released. You can try to modify your Podfile to point to the master branch and retest. I tested it and now it's OK.

```ruby
pod 'SDWebImage',  git:'git@github.com:rs/SDWebImage.git', :branch => 'master'
```

**Comment by dreampiggy** [comment]
@bpoplauschi Any plan to release another patch version for 4.2 :) ?

**Comment by bpoplauschi** [comment]
Got it @dreampiggy. I will release 4.2.2 today.

**Comment by bpoplauschi** [comment]
Fixed by #2093

**Comment by bpoplauschi** [comment]
4.2.2 is released
